Session7/SparseMatrix.java

```diff
@@ -0,0 +1,177 @@
+/**
+* This class is designed to make a sparse matrix.
+* 
+* @author Aman Gautam
+* 
+*/
+public final class SparseMatrix {
+
+	private int[][] sparseMatrix;
+	private int rows;
+	private int columns;
+	
+	private static final int VALUE = 0;
+    private static final int ROW_NUMBER = 1;
+    private static final int COL_NUMBER = 2;
+	
+    /*
+     * This class represents the Sparse matrix where most
+     * of the elements are zero .
+     */
+	public SparseMatrix(int[][] givenArray) {
+		int length = 0;
+		int maxColLength = 0;														
+		
+		if(givenArray == null) {
+			throw new AssertionError("Cant be null.");
+		}
+		
+		if(givenArray.length == 0) {
+		    throw new AssertionError("Cant be empty.");
```

Comment:
The message should be more meaningful. Try to use `try` and `catch` block to handle the error state also if requires